addons/stock_picking_batch/models/stock_move_line.py

```diff
@@ -88,3 +90,253 @@ def _add_to_wave(self, wave=False):
         if picking_to_wave_vals_list:
             self.env['stock.picking'].create(picking_to_wave_vals_list)
         wave.action_confirm()
+
+    def _is_auto_waveable(self):
+        self.ensure_one()
+        if not self.picking_id \
+           or self.picking_id.state != 'assigned' \
+           or float_is_zero(self.quantity, precision_rounding=self.product_uom_id.rounding) \
+           or self.batch_id.is_wave \
+           or not self.picking_type_id._is_auto_wave_grouped() \
+           or (self.picking_type_id.wave_group_by_category and self.product_id.categ_id not in self.picking_type_id.wave_category_ids):  # noqa: SIM103
```

Comment:
It's annoying to have a linter override in the code, but imo it's also less readable to return just the `if` condition instead of `False`. I defer to the final reviewer for whichever they prefer